sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -55,6 +55,9 @@ static const struct snd_soc_dapm_route skl_hda_map[] = {
 	{ "Digital CPU Capture", NULL, "Digital Codec Capture" },
 	{ "codec2_in", NULL, "Alt Analog CPU Capture" },
 	{ "Alt Analog CPU Capture", NULL, "Alt Analog Codec Capture" },
+
+	{"dmic01", NULL, "DMIC AIF"},
+	{"dmic02", NULL, "DMIC AIF"},
```

Comment:
Do we need this? It seems there is no dmic widget in the topology.

Reply:
same, not sure if this is needed for SOF, and it's not supported by the SKL/cAVS driver.

Reply:
Actually, this is the widget in dmic codec driver, so we need to connect to it, just like "Alt Analog CPU Capture" which is defined in hdac_hda codec.
````
static const struct snd_soc_dapm_route intercon[] = {
	{"DMIC AIF", NULL, "DMic"},
};

````

Reply:
@RanderWang Yes, "DMIC AIF" is from dmic codec driver, but where is "dmic01" and "dmic02"?

Reply:
dmic01 and dmic02 is created by topology